python-package/lightgbm/basic.py

```diff
@@ -811,10 +811,10 @@ def _data_from_pandas(
                 data[col] = data[col].cat.set_categories(category)
     if len(cat_cols):  # cat_cols is list
         data[cat_cols] = data[cat_cols].apply(lambda x: x.cat.codes).replace({-1: np.nan})
-    if categorical_feature == 'auto':  # use cat cols from DataFrame
+
+    # use cat cols from DataFrame
+    if categorical_feature == 'auto':
         categorical_feature = cat_cols_not_ordered
-    else:  # use cat cols specified by user
-        categorical_feature = list(categorical_feature)  # type: ignore[assignment]
```

Comment:
`categorical_feature` can only be a list of strings, a list of integers, or the literal string `'auto'`.

So I don't think this `else:` clause is helpful. When it's reached, under correctly-provided inputs it's just calling `list()` on something that's already a list, which has no effect.

I guess this could be providing a small bit of help for cases where some other type like a tuple was passed in, but if we want to support that then:

* it should be done at all public interfaces that support `categorical_feature`
* it shouldn't be buried inside a `pandas`-specific function

I think this can and should be removed, in the interest of further simplifying this function.